### PR TITLE
Advanced filter changes: Namespace input ids, Place input before label

### PR DESCRIPTION
**Pull request by josh-schlick**
Associate the input to the label implicityly
- There was an issue when trying to place multiple tables on a single page where clicking a label would toggle the wrong checkbox. This fixes the problem.

### CODE REVIEW

**Comment by ernieturner** [comment]
Looks good to me. Do you have any sense on how breaking of a change this would be? I get the swap of the label/checkbox, but wasn't sure how breaking of a change it'd be with the CSS. Just trying to figure out what version we should tag this with. Also, you might as well push up a change to the version in the package.json file.

**Comment by josh-schlick** [comment]
I think this would have a high probability of breaking code if someone had overwritten any of the advanced-filter css.  Otherwise, I don't think this would break anything.

**Comment by josh-schlick** [comment]
0.5.6 or 0.6.0?

**Comment by ernieturner** [comment]
Lets just go with 0.6.0 to be safe. We really need to just make the decision to move this library to 1.0.0 and go from there.

File: src/js/table/BaseTable.js [hunk 1]
```diff
@@ -177,10 +177,11 @@ module.exports = {
      * @returns {ReactElement} - A React div element containing an advanced filter item.
      */
     getAdvancedFilterItemMarkup: function(filter, index) {
+        var filterId = `filter-${this.props.componentId}-${index}`;
         return (
             <div key={index} className="advanced-filter-item">
-                <label htmlFor={`filter-${index}`} className="no-select">{filter.label}</label>
-                <input id={`filter-${index}`} type="checkbox" checked={filter.checked || false} onChange={this.handleAdvancedFilterToggle.bind(this, filter)} />
+                <input id={filterId} type="checkbox" checked={filter.checked || false} onChange={this.handleAdvancedFilterToggle.bind(this, filter)} />
+                <label htmlFor={filterId} className="no-select">{filter.label}</label>
```
Comment (jasonjohnson115) [inline src/js/table/BaseTable.js]: I'd consider simply putting the input inside the label. The label can be placed on either side of the control. There are a number of benefits to doing this ranging from not needing an id on every input to the htmlFor attribute not being necessary. See associating a label with a control explicitly vs implicitly: https://www.w3.org/TR/html401/interact/forms.html#h-17.9.1

I sort of feel that row-reverse flex is an awkward solution for an already solved problem.

**Comment by josh-schlick** [comment]
good idea.  addressed

**Pull request merged**